org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/BaseDocumentLifeCycleHandler.java

```diff
@@ -258,6 +259,10 @@ public IStatus validateDocument(String uri, boolean debounce, IProgressMonitor m
 		}
 
 		toValidate.add(unit);
+		if (!unit.equals(sharedASTProvider.getActiveJavaElement())) {
```

Comment:
Right. Looks like this is needed because switching opened tabs doesn't trigger a `didOpen`, so the active element is stale. Is that correct ?

Reply:
Yes, it is.